src/csharp/Microsoft.Spark/ML/Feature/Bucketizer.cs

```diff
@@ -0,0 +1,206 @@
+        }
+        
+        /// <summary>
+        /// The list of columns that the `Bucketizer` will create in the DataFrame.
+        /// </summary>
+        /// <param name="value">List of column names which will contain the bucket ID</param>
+        /// <returns>`Bucketizer`</returns>
+        public Bucketizer SetOutputCols(List<string> value)
+        {
+            return WrapAsBucketizer(_jvmObject.Invoke("setOutputCols", value));
+        }
+        
+        /// <summary>
+        /// Executes the `Bucketizer` and transforms the DataFrame to include the new column or
+        /// columns with the bucketed data.
+        /// </summary>
+        /// <param name="source">The DataFrame to add the bucketed data to</param>
+        /// <returns>`DataFrame` containing the original data and the new bucketed columns</returns>
+        public DataFrame Transform(DataFrame source)
+        {
+            return new DataFrame((JvmObjectReference)_jvmObject.Invoke("transform"
+                , source));
+        }
+
+        /// <summary>
+        /// The reference we get back from each call isn't usable unless we wrap it in a new dotnet
+        ///  `Bucketizer`
+        /// </summary>
+        /// <param name="obj">The `JvmObjectReference` to convert into a dotnet `Bucketizer`</param>
+        /// <returns>`Bucketizer`</returns>
+        private static Bucketizer WrapAsBucketizer(object obj)
+        {
+            return new Bucketizer((JvmObjectReference)obj);
+        }
+
+        /// <summary>
+        /// The uid that was used to create the `Bucketizer`. If no `UID` is passed in when creating
+        ///  the `Bucketizer` then a random `UID` is created when the `Bucketizer` is created.
+        /// </summary>
+        /// <returns>string `UID` identifying the `Bucketizer`</returns>
+        public string Uid()
+        {
+            return (string)_jvmObject.Invoke("uid");
+        }
+
+        /// <summary>
+        /// How should the `Bucketizer` handle invalid data, choices are "skip", "error" or "keep"
+        /// </summary>
+        /// <returns>`BucketizerInvalidOptions`</returns>
+        public BucketizerInvalidOptions GetHandleInvalid()
+        {
+            string handleInvalid = (string)_jvmObject.Invoke("getHandleInvalid");
+            if (BucketizerInvalidOptions.TryParse(handleInvalid, true, 
+                out BucketizerInvalidOptions result))
+            {
+                return result;
+            }
+            
+            return result;
+        }
+
+        /// <summary>
+        /// Tells the `Bucketizer` what to do with invalid data.
+        ///
+        /// Choices are "skip", "error" or "keep". Default is "error"
+        /// </summary>
+        /// <param name="value">`BucketizerInvalidOptions`, "skip", "error" or "keep"</param>
+        /// <returns>`Bucketizer`</returns>
+        public Bucketizer SetHandleInvalid(BucketizerInvalidOptions value)
```

Comment:
```suggestion
        public Bucketizer SetHandleInvalid(string value)
```

Let's let spark handle validating the passed in values.